### DISCUSSION TITLE
How can I get CollectionView to scroll in both directions, due to a long and wide list.

### DISCUSSION THREAD

**Discussion by svn2208** [dotnet/maui#21228] [Q&A]
Hi there,

I need to display a large amount of text (Logfile). I structured it in three colums (time, xx, long text-line). I am not able to convince the CollectionView to scroll down, to see the hundreds of thousands lines of the log, but also the scroll to the right to see the end of the very long lines. 

I can only archive to scroll down, but not to the right.

`        <CollectionView x:Name="TelegramViewer"
                        Grid.Row="0"
                        ItemSizingStrategy="MeasureAllItems"
                        SelectionMode="Multiple">
            <CollectionView.ItemTemplate>
                <DataTemplate>
                    <Grid Padding="1" CompressedLayout.IsHeadless="True">
                        <Grid.ColumnDefinitions>
                            <ColumnDefinition Width="220" />
                            <ColumnDefinition Width="45" />
                            <ColumnDefinition  />
                        </Grid.ColumnDefinitions>
                        <Grid.RowDefinitions>
                            <RowDefinition Height="25" />
                        </Grid.RowDefinitions>
                        <Label Text="{Binding Time, Mode=OneTime}" HeightRequest="25" Padding="10,0,0,0"/>
                        <Label Grid.Column="1" Text="{Binding TrainId, Mode=OneTime}" HeightRequest="25"/>
                        <Label Grid.Column="2" Text="{Binding Converter={StaticResource messageConverter}, Mode=OneTime}" LineBreakMode="NoWrap" HeightRequest="25"/>
                    </Grid>
                </DataTemplate>
            </CollectionView.ItemTemplate>
        </CollectionView>`

**Comment by sjordanGSS** [comment]
I don't think it's possible to do this, but maybe you could wrap your message label in a ScrollView so log lines can be scrolled individually?
